egs/mobvoi/v1/local/chain/tuning/run_tdnn_1a.sh

```diff
@@ -0,0 +1,293 @@
+done
+
+if [ $stage -le 1 ]; then
+  echo "$0: creating lang directory $lang with chain-type topology"
+  # Create a version of the lang/ directory that has one state per phone in the
+  # topo file. [note, it really has two states.. the first one is only repeated
+  # once, the second one has zero or more repeats.
+  if [ -d $lang ]; then
+    if [ $lang/L.fst -nt data/lang/L.fst ]; then
+      echo "$0: $lang already exists, not overwriting it; continuing"
+    else
+      echo "$0: $lang already exists and seems to be older than data/lang ..."
+      echo " ... not sure what to do.  Exiting."
+      exit 1;
+    fi
+  else
+    rm -rf $lang
+    cp -r data/lang $lang
+    silphonelist=$(cat $lang/phones/silence.csl) || exit 1;
+    nonsilphonelist=$(cat $lang/phones/nonsilence.csl) || exit 1;
+    # Use our special topology... note that later on may have to tune this
+    # topology.
+    local/gen_topo.pl 4 1 $nonsilphonelist $silphonelist >$lang/topo
+  fi
+fi
+
+if [ $stage -le 2 ]; then
+  # Get the alignments as lattices (gives the chain training more freedom)
+  # use the same num-jobs as the alignments
+  steps/align_fmllr_lats.sh --nj 75 --cmd "$train_cmd" ${lores_train_data_dir} \
+    data/lang $gmm_dir $lat_dir
+  rm $lat_dir/fsts.*.gz # save space
+fi
+
+if [ $stage -le 3 ]; then
+  local/copy_lat_dir.sh --nj 75 --utt-prefixes "$aug_prefix" \
+    $combined_train_data_dir $lat_dir $combined_lat_dir
+fi
+
+if [ $stage -le 4 ]; then
+  # Build a tree using our new topology.  We know we have alignments for the
+  # speed-perturbed data (local/nnet3/run_ivector_common.sh made them), so use
+  # those.  The num-leaves is always somewhat less than the num-leaves from
+  # the GMM baseline.
+  if [ -f $tree_dir/final.mdl ]; then
+    echo "$0: $tree_dir/final.mdl already exists, refusing to overwrite it."
+    exit 1;
+  fi
+  local/chain/build_tree.sh \
+    --frame-subsampling-factor 3 \
+    --cmd "$train_cmd" ${lores_train_data_dir} \
+    $lang $ali_dir $tree_dir
+
+  echo "$0: Creating an unnormalized phone language model for the denominator graph..."
+  id_sil=`cat data/lang/phones.txt | grep "SIL" | awk '{print $2}'`
+  id_word=`cat data/lang/phones.txt | grep "hixiaowen" | awk '{print $2}'`
+  id_freetext=`cat data/lang/phones.txt | grep "freetext" | awk '{print $2}'`
+  cat <<EOF > $tree_dir/phone_lm.txt
+0 1 $id_sil $id_sil
+1 2 $id_word $id_word
+2 3 $id_sil $id_sil
+1 4 $id_freetext $id_freetext
+4 5 $id_sil $id_sil
+3 1.9
+5 0.7
+EOF
+  fstcompile $tree_dir/phone_lm.txt $tree_dir/phone_lm.fst
+  fstdeterminizestar $tree_dir/phone_lm.fst $tree_dir/phone_lm.fst.tmp
+  mv $tree_dir/phone_lm.fst.tmp $tree_dir/phone_lm.fst
+fi
+
+if [ $stage -le 5 ]; then
+  echo "$0: creating neural net configs using the xconfig parser";
+  num_targets=$(tree-info $tree_dir/tree | grep num-pdfs | awk '{print $2}')
+  learning_rate_factor=$(echo "print 0.5/$xent_regularize" | python)
```

Comment:
@freewym 

I guess you are using python**2** locally. But these days not all people are using python2.

```
$(echo "print 0.5/$xent_regularize" | python)
```
is not valid for python3 and results in syntax errors.

I suggest using `print(0.5/$xent_regularize)`, that is, use braces here, for better
portability.

A better way is to always use ``python2`` or ``python3`` explicitly.